plugins/JsTrackerInstallCheck/JsTrackerInstallCheck.php

```diff
@@ -0,0 +1,149 @@
+
+        $hasInstallCheckParam = $request->hasParam(self::QUERY_PARAM_NAME);
+        if (!$hasInstallCheckParam) {
+            return;
+        }
+
+        $trackerInstallCheckParam = $request->getParams()[self::QUERY_PARAM_NAME];
+        if (empty($trackerInstallCheckParam)) {
+            return;
+        }
+
+        // If the request has already been excluded, we don't want to override that
+        if (empty($excluded)) {
+            $excluded = 'excluded: testing whether JS tracker is installed';
+        }
+        StaticContainer::get(LoggerInterface::class)->debug('Excluding visit as JS tracker install test.');
+
+        $nonceOptionString = Option::get(self::OPTION_NAME_PREFIX . $request->getIdSite());
+        if (empty($nonceOptionString)) {
+            return;
+        }
+
+        $nonceOptionArray = json_decode($nonceOptionString, true);
+        if (empty($nonceOptionArray)) {
+            return;
+        }
+
+        // Make sure that the nonce matches
+        if (empty($nonceOptionArray['nonce']) || $nonceOptionArray['nonce'] !== $trackerInstallCheckParam) {
+            return;
+        }
+
+        // If the nonce is older 30 seconds, ignore it. This should be plenty of time because an API call creates the nonce just before opening the site
+        if (empty($nonceOptionArray['time']) || Date::getNowTimestamp() - $nonceOptionArray['time'] > 30) {
+            return;
+        }
+
+        // Since the nonce matches and hasn't expired, update the option indicating success
+        $nonceOptionArray['isSuccessful'] = true;
+        Option::set(self::OPTION_NAME_PREFIX . $request->getIdSite(), json_encode($nonceOptionArray));
+    }
+
+    /**
+     * Check whether a test request has been recorded for the provided nonce. If no nonce is provided, the recorded
+     * result for the site will be returned.
+     *
+     * @param string $idSite
+     * @param string $nonce The unique nonce used to identify the test requests. Optionally can be left empty if simply
+     * wanting to check if the site has been successfully tested.
+     * @return array list of containers ['isSuccess' => true]
+     */
+    public function checkForJsTrackerInstallTestSuccess(string $idSite, string $nonce = ''): array
+    {
+        $nonceOptionString = Option::get(self::OPTION_NAME_PREFIX . $idSite);
+        if (empty($nonceOptionString)) {
+            return ['isSuccess' => false];
+        }
+
+        $nonceOptionArray = json_decode($nonceOptionString, true);
+        if (empty($nonceOptionArray)) {
+            return ['isSuccess' => false];
+        }
+
+        // Check if the nonce matches the recorded nonce
+        if (!empty($nonce) && (empty($nonceOptionArray['nonce']) || $nonceOptionArray['nonce'] !== $nonce)) {
+            return ['isSuccess' => false];
+        }
+
+        return ['isSuccess' => !empty($nonceOptionArray['isSuccessful'])];
```

Comment:
Where is the sense in returning an array here? It always holds one element with the same key, so you also could directly return it's value.

Reply:
Good point. That's left over from me putting this method in the API class and then deciding to move it to this class. I'll change the return type to boolean.